common/src/main/java/org/corfudb/common/protocol/API.java

```diff
@@ -47,4 +62,118 @@ public static Request newPingRequest(Header header) {
+                .setPingResponse(pingResponse)
+                .build();
+    }
+
+    public static Request newAuthenticateRequest(Header header, UUID clientId, UUID serverId) {
+        AuthenticateRequest authRequest = AuthenticateRequest.newBuilder()
+                                            .setClientId(getUUID(clientId))
+                                            .setServerId(getUUID(serverId))
+                                            .build();
+        return Request.newBuilder()
+                .setHeader(header)
+                .setAuthenticateRequest(authRequest)
+                .build();
+    }
+
+    public static Response newAuthenticateResponse(Header header, UUID serverId, String version) {
+        AuthenticateResponse authResponse = AuthenticateResponse.newBuilder()
+                                                .setServerId(getUUID(serverId))
+                                                .setCorfuVersion(version)
+                                                .build();
+        return Response.newBuilder()
+                .setHeader(header)
+                .setAuthenticateResponse(authResponse)
+                .build();
+    }
+
+    public static Request newGetLayoutRequest(Header header, long epoch) {
+        GetLayoutRequest layoutRequest = GetLayoutRequest.newBuilder().setEpoch(epoch).build();
+        return Request.newBuilder()
+                .setHeader(header)
+                .setGetLayoutRequest(layoutRequest)
+                .build();
+    }
+
+    public static Request newPrepareLayoutRequest(Header header,long epoch, long rank) {
+        PrepareLayoutRequest prepLayoutRequest = PrepareLayoutRequest.newBuilder()
+                                                            .setEpoch(epoch)
+                                                            .setRank(rank)
+                                                            .build();
+        return Request.newBuilder()
+                .setHeader(header)
+                .setPrepareLayoutRequest(prepLayoutRequest)
+                .build();
+    }
+
+    public static Request newQueryStreamRequest(Header header, QueryStreamRequest.ReqType type,
+                                                List<StreamAddressRange> ranges) {
+        QueryStreamRequest qsRequest = QueryStreamRequest.newBuilder()
+                                                .setType(type)
+                                                .addAllStreamRanges(ranges)
+                                                .build();
+        return Request.newBuilder()
+                .setHeader(header)
+                .setQueryStreamRequest(qsRequest)
+                .build();
+    }
+
+    public static Request newQueryStreamRequest(Header header, List<StreamAddressRange> ranges) {
+        return newQueryStreamRequest(header, QueryStreamRequest.ReqType.STREAMS, ranges);
+    }
+
+    public static Request newRestartRequest(Header header) {
```

Comment:
Since these are static, I suggest we drop the new* prefix, it seems redundant.

Reply:
Agreed. I will change them to `get*` as in `getRestartRequest()`.